src/_pytest/capture.py

```diff
@@ -177,7 +177,7 @@ def name(self) -> str:
     def mode(self) -> str:
         # TextIOWrapper doesn't expose a mode, but at least some of our
         # tests check it.
-        return self.buffer.mode.replace("b", "")
+        return self.buffer.mode.replace("b", "")  # type: ignore[no-any-return,attr-defined]
```

Comment:
The comment anticipated the issue before mypy found it so I guess it's fine to noqa.

Reply:
Per https://github.com/python/typeshed/blob/ca65e08/stdlib/_io.pyi#L155C25-L155C36, `self.buffer` is of type `_BufferT_co = TypeVar("_BufferT_co", bound=_WrappedBuffer, default=_WrappedBuffer, covariant=True)` which is a protocol that doesn't specify `mode`: https://github.com/python/typeshed/blob/ca65e08/stdlib/_io.pyi#L118-L139.

I wonder, however, if doing something like
```suggestion
        assert hasattr(self.buffer, 'mode')
        return cast(str, self.buffer.mode.replace("b", ""))
```
would help MyPy figure out the actual typing.

Reply:
Let's try